File: user_pgm.py
import sys
import fileinput

def get_input():
    input = fileinput.input()[0].split()
    input[2] = int(input[2])
    return input

def print_char(to_send):
    print(to_send.lower())
    sys.exit()

# You should write your code in this function. input is a list of 3 items. input[0] is
# the current state. This is in the form of a string in which each character is either
# a '_' character if the letter has not been guessed or the correct character. Example:
# if the word to be guessed is "apple" and you have previously guessed 'a' and 'l', then
# state will appear as "a__l_"
def user_func(input):
    # YOUR CODE HERE

    
to_send = ''

print_char(user_func(get_input()))

File: user_pgm.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

void get_in(char* state, char* guesses, int* x) {
  scanf("%s", state);
  scanf("%s", guesses);
  scanf("%d", x);

  if (*x == 0) {
    *x = atoi(guesses);
    memset(&guesses[0], 0, sizeof(guesses));
  }
}

void print_char(char to_send) {
  printf("%c", tolower(to_send));

  exit(0);
}

/* 
 * This function should contain all of your code. It gets the strings state and guesses
 * and the int num_guesses as parameters. state is a string representing the current
 * state of the secret word. Any characters in the string that have not been correctly
 * guessed yet are represented as '_'. Example: if the word to be guessed is "apple",
 * and you have previously guessed 'a' and 'l', the string will be "a__l_" (ending with
 * a null terminating byte). guesses is a character array of the previously guessed
 * characters ending in a null terminating byte. The return value should be the
 * character that you want to guess.
 */
char user_func(char* state, char* guesses, int num_guesses) {
  /* YOUR CODE HERE */
}

int main(void) {
  char state[63], guesses[63], to_send;
  int num_guesses;

  get_in(state, guesses, &num_guesses);

  print_char(user_func(state, guesses, num_guesses));
  return 0;
}
